tests/utils/test_utils.go

```diff
@@ -0,0 +1,433 @@
+			} else {
+				devObj := parser.DevfileObj{
+					Ctx:  ctx,
+					Data: newDevfile,
+				}
+				err = devObj.WriteYamlDevfile()
+				if err != nil {
+					LogErrorMessage(fmt.Sprintf("Writing devfile : %v", err))
+				} else {
+					devfile.SchemaParsed = false
+				}
+			}
+
+		}
+	} else {
+		LogInfoMessage(fmt.Sprintf("Marshall and write devfile %s", devfile.FileName))
+		c, marshallErr := yaml.Marshal(&(devfile.SchemaDevFile))
+
+		if marshallErr != nil {
+			err = errors.New(LogErrorMessage(fmt.Sprintf("Marshall devfile %s : %v", devfile.FileName, marshallErr)))
+		} else {
+			err = ioutil.WriteFile(fileName, c, 0644)
+			if err != nil {
+				LogErrorMessage(fmt.Sprintf("Write devfile %s : %v", devfile.FileName, err))
+			} else {
+				devfile.SchemaParsed = false
+			}
+		}
+	}
+	return err
+}
+
+// Use the parser to parse a devfile on disk
+func (devfile *TestDevfile) parseSchema() error {
+
+	var err error
+	if !devfile.SchemaParsed {
+		LogInfoMessage(fmt.Sprintf("Parse and Validate %s : ", devfile.FileName))
+		devfile.ParsedSchemaObj, err = devfilepkg.ParseAndValidate(devfile.FileName)
+		if err != nil {
+			LogErrorMessage(fmt.Sprintf("From ParseAndValidate %v : ", err))
+		}
+		devfile.SchemaParsed = true
+	}
+	return err
+}
+
+// Verify verifies the contents of the specified devfile with the expected content
+func (devfile TestDevfile) Verify() error {
+
+	LogInfoMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	var errorString []string
+
+	err := devfile.parseSchema()
+
+	if err != nil {
+		errorString = append(errorString, LogErrorMessage(fmt.Sprintf("parsing schema %s : %v", devfile.FileName, err)))
+	} else {
+		LogInfoMessage(fmt.Sprintf("Get commands %s : ", devfile.FileName))
+		commands, _ := devfile.ParsedSchemaObj.Data.GetCommands(common.DevfileOptions{})
+		if commands != nil && len(commands) > 0 {
+			err = devfile.VerifyCommands(commands)
+			if err != nil {
+				errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+			}
+		} else {
+			LogInfoMessage(fmt.Sprintf("No command found in %s : ", devfile.FileName))
+		}
+	}
+
+	if err == nil {
+		LogInfoMessage(fmt.Sprintf("Get components %s : ", devfile.FileName))
+		components, _ := devfile.ParsedSchemaObj.Data.GetComponents(common.DevfileOptions{})
+		if components != nil && len(components) > 0 {
+			err = devfile.VerifyComponents(components)
+			if err != nil {
+				errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+			}
+		} else {
+			LogInfoMessage(fmt.Sprintf("No components found in %s : ", devfile.FileName))
+		}
+	}
+
+	var returnError error
+	if len(errorString) > 0 {
+		returnError = errors.New(fmt.Sprint(errorString))
+	}
+	return returnError
+
+}
+
+// EditCommands modifies random attributes for each of the commands in the devfile.
+func (devfile TestDevfile) EditCommands() error {
```

Comment:
should this be in command utils file?

Reply:
I prefer where it is because the use the function makes of the parser, wanted to keep the command_utils pure command stuff.